esp/esp/program/modules/handlers/studentregphasezero.py

```diff
@@ -0,0 +1,220 @@
+  527 Franklin St, Cambridge, MA 02139
+  Phone: 617-379-0178
+  Email: web-team@learningu.org
+"""
+
+from esp.utils.web import render_to_response
+from esp.middleware.threadlocalrequest import get_current_request
+from esp.program.modules.base import ProgramModuleObj, main_call, aux_call, meets_deadline, needs_student, meets_grade, meets_cap, no_auth, needs_admin
+from esp.users.models import Record, ESPUser, Permission
+from esp.program.models import PhaseZeroRecord
+from esp.program.modules.forms.phasezero import SubmitForm
+from esp.dbmail.models import send_mail
+from esp.tagdict.models import Tag
+
+from django.conf import settings
+from django.template.loader import render_to_string
+from django.contrib.auth.models import Group
+from django.db.models.query import Q
+
+import random, copy, datetime, re
+
+class StudentRegPhaseZero(ProgramModuleObj):
+    def students(self, QObject = False):
+        q_phasezero = Q(phasezerorecord__program=self.program)
+
+        if QObject:
+            retVal = {'phasezero': q_phasezero}
+            return retVal
+
+        retVal = {'phasezero': ESPUser.objects.filter(q_phasezero).distinct()}
+        return retVal
+
+    def studentDesc(self):
+        retVal = {'phasezero': """Students who have entered the Student Lottery."""}
+        return retVal
+
+    def isCompleted(self):
+        return get_current_request().user.canPassPhaseZero(self.program)
+
+    @classmethod
+    def module_properties(cls):
+        return {
+            "admin_title": "Student Registration Phase Zero",
+            "link_title": "Student Registration Phase Zero",
+            "module_type": "learn",
+            "seq": 2,
+            "required": True
+        }
+
+    @main_call
+    @needs_student
+    @meets_grade
+    def studentregphasezero(self, request, tl, one, two, module, extra, prog):
+        """
+        Serves the Phase Zero student reg page. The initial page includes a button
+        to enter the student lottery. Following entering the lottery, students are
+        served a confirmation page and lottery number form (for self-assigning groups).
+        """
+        context = {}
+        context['program'] = prog
+        user = request.user
+        in_lottery = PhaseZeroRecord.objects.filter(user=user, program=prog).exists()
+
+        op = ''
+        join_error = False
+        if 'op' in request.POST:
+            op = request.POST['op']
+        if op == 'join':
+            if len(request.POST['student_selected'].strip()) == 0:
+                join_error = 'Error - You must select a student\'s username.'
+
+            else:
+                join_user = request.POST['student_selected']
+                group = PhaseZeroRecord.objects.filter(user=join_user, program=prog)[0]
+                if in_lottery:
+                    old_group = PhaseZeroRecord.objects.filter(user=user, program=prog)[0]
+                users = group.user.all()
+                num_users = len(users)
```

Comment:
Are you actually using `users` anywhere?  If not, just do `group.user.count()`, as below.